### PR TITLE
React Native 0.69.4 Upgrade

### PR DESCRIPTION
**Pull request by geriux**
This PR adds support for React Native `0.69.4`

The main changes are:

- Updating `bin/generate-podspecs.sh` to support the new changes like Hermes being bundled with the React Native library.
- Updating the script for `bundle:android:bytecode` since the path for Hermes changed.
- Add a new `clean:gutenberg:distclean` postinstall script to clean up `node_modules` in Gutenberg since it was generating a conflict issue with multiple versions of React. There will be a follow-up ticket to investigate this further so we can remove this extra step.
- Updates the third-party podspecs.

### Feature branch

- [Gutenberg](https://github.com/WordPress/gutenberg/pull/43485)

### Android integration

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5170)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/43486)
- [WordPress Android](https://github.com/wordpress-mobile/WordPress-Android/pull/17066)

### iOS integration

- [Gutenberg Mobile](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5192)
- [Gutenberg](https://github.com/WordPress/gutenberg/pull/44889)
- [WordPress iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/19326)

To test:

- CI checks must pass
- Manual testing was done internally

PR submission checklist:

- [ ] I have considered adding unit tests where possible.
- [ ] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: package.json
Comment (fluiddot) [inline package.json:57]: I'm wondering if it's necessary cleaning up Gutenberg on every dependency install 🤔. I noticed in the `package-lock.json` that the `react` dependency has been updated to 18 ([reference](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5193/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519R8289-R8290)), shouldn't that be enough to install the correct versions of the dependencies under Gutenberg?

My main concern about this workaround is that I understand that it will increase severely the duration of the dependencies install command, do we know roughly how much time we expect that to be increased?

Comment (geriux) [reply package.json:57]: Hey there, sorry for taking a bit to reply to this comment 😅 

>shouldn't that be enough to install the correct versions of the dependencies under Gutenberg?

It looks like we have other dependencies that still use React 17 and it's causing conflicts with the dependencies and the editor.

>My main concern about this workaround is that I understand that it will increase severely the duration of the dependencies install command, do we know roughly how much time we expect that to be increased?

What do you mean by: it will increase severely? That new step we are adding it's just removing any dependencies that might have been installed in Gutenberg when we run `npm install` in `Gutenberg mobile` before running `npm ci`. It's just a few seconds while it removes all of the `node_module` folders within `Gutenberg` that might have com from running `npm install` from the parent folder. From all of the tests I've done while working with the upgrade, it doesn't take more than a few seconds.

We could add a follow-up task to investigate further the dependencies we have to avoid this workaround.

Comment (fluiddot) [reply package.json:57]: > It looks like we have other dependencies that still use React 17 and it's causing conflicts with the dependencies and the editor.

Ah, interesting. Do we know which dependencies are still using React 17?

On the other hand, do we need to run `npm run clean:gutenberg:distclean` everytime or is this a one-off?

> What do you mean by: it will increase severely? That new step we are adding it's just removing any dependencies that might have been installed in Gutenberg when we run npm install in Gutenberg mobile before running npm ci. It's just a few seconds while it removes all of the node_module folders within Gutenberg that might have com from running npm install from the parent folder. From all of the tests I've done while working with the upgrade, it doesn't take more than a few seconds.

Ah, ok, my concern came because I thought that cleaning Gutenberg would increase the duration of the `npm ci --prefix gutenberg` command. In case it's just adding a few seconds then it's ok to continue with this workaround. Thanks for elaborating on this 🙇 !

> We could add a follow-up task to investigate further the dependencies we have to avoid this workaround.

This would be great! Following on your comment regarding the fact that the duration won't be increased, I understand that is not a high priority, but having a ticket as a follow-up would be helpful.

Comment (geriux) [reply package.json:57]: > Ah, interesting. Do we know which dependencies are still using React 17?

I saw a few but can't remember from when I tested, I'll add this in the ticket as well so we have a list.

> On the other hand, do we need to run npm run clean:gutenberg:distclean everytime or is this a one-off?

Just when an `npm install` happens, I know it's not ideal but we can work on this after this gets merged.

> This would be great! Following on your comment regarding the fact that the duration won't be increased, I understand that is not a high priority, but having a ticket as a follow-up would be helpful.

I mean it will be increased by a few seconds but it won't impact our development process or CI so much. But still, I'll create a ticket to investigate this a bit further.

**Comment by geriux** [comment]
FYI some CI checks will fail until this PR https://github.com/WordPress/gutenberg/pull/46220 gets merged.

**Comment by fluiddot** [comment]
> FYI some CI checks will fail until this PR [WordPress/gutenberg#46220](https://github.com/WordPress/gutenberg/pull/46220) gets merged.

Heads up that https://github.com/WordPress/gutenberg/pull/46220 is already merged in `trunk`.

**Review by derekblank — APPROVED** [review]
LGTM! 🚀

**Comment by fluiddot** [comment]
> I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/trunk/RELEASE-NOTES.txt) if necessary.

@geriux @derekblank please when you're available it would be great to include this change in the release notes. I don't think we should block the merge due to this, especially as we'd need to have `trunk` with the RN upgrade changes to match Gutenberg's `trunk` changes. Hence, let's merge this PR and update the release notes in a separate PR 👍 .

**Pull request merged**
